### PR TITLE
allow specifying lower disks for Walk()

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
allow specifying lower disks for Walk()

## Motivation and Context
useful when you may want Walk() with
reduced quorum requirements.

This is a requirement for https://github.com/minio/minio/pull/17792

## How to test this PR?
This was an ask by @krisis , this will be mostly useful with batch-expire PR

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by krisis — APPROVED** [review]
Neat!

File: cmd/object-api-interface.go [hunk 1]
```diff
@@ -97,6 +97,8 @@ type ObjectOptions struct {
 
 	WalkFilter      func(info FileInfo) bool // return WalkFilter returns 'true/false'
 	WalkMarker      string                   // set to skip until this object
+	WalkLatestOnly  bool                     // returns only latest versions for all matching objects
+	WalkAskDisks    string                   // dictates how many disks are being listed
```
Comment (klauspost) [inline cmd/object-api-interface.go:101]: None of this is being tested right now. You never set either. We shouldn't add a bunch of code that is never hit.

**Review by klauspost — CHANGES_REQUESTED** [review]
Not sure if you omitted where you want to use this.

File: cmd/object-api-interface.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/object-api-interface.go:101]: This is actually used in the batch-expire feature.

https://github.com/miniohq/minio/tree/RELEASE.2023-08-04T17-40-21Z.hotfix

Comment (harshavardhana) [reply cmd/object-api-interface.go:101]: I sent it separately but it can be merged along with batch expire if you insist.

Comment (harshavardhana) [reply cmd/object-api-interface.go:101]: @klauspost ^^

Comment (klauspost) [reply cmd/object-api-interface.go:101]: Please add it so we can get the complete picture. I presume it needs this anyway.

Comment (harshavardhana) [reply cmd/object-api-interface.go:101]: It can't yet be added since this is a dependent PR for https://github.com/minio/minio/pull/17792

**Comment by harshavardhana** [comment]
PTAL @poornas

**Review by poornas — APPROVED** [review]

**Pull request merged**